### PR TITLE
EQ-413 Transaction ID

### PR DESCRIPTION
**Pull request by warrenbailey**
### What is the context of this PR?

Adding Transaction ID to the Metadata store, and submitted response downstream and also updating logging events
### How to review
1. Check out this branch
2. Run the application
3. Check that the transaction ID is being written to the log files at info level
4. Complete a survey and check that the transaction id is displayed on the thank you page
- [ ] Do all commits have sensible commit messages?
- [ ] Is all new code unit tested?
- [ ] Are there integration tests if needed?
- [ ] Is there sufficient logging?
- [ ] Is there documentation or is the code self-describing?
### Who worked on the PR

@warrenbailey and @bennpearson

### CODE REVIEW

**Comment by warrenbailey** [comment]
Just waiting for design input

File: app/main/views/questionnaire.py [hunk 1]
```diff
@@ -41,7 +42,9 @@ def survey(eq_id, collection_id, location):
             next_location = questionnaire_manager.get_current_location()
 
             current_user.save()
+            metadata = MetaDataStore.get_instance(current_user)
 
+            logger.info("Redirecting user to next location %s with tx_id= %s", next_location, metadata.tx_id)
```
Comment (collisdigital) [inline app/main/views/questionnaire.py]: Remove space after =

**Comment by warrenbailey** [comment]
All done, just needs front end work on the thank you page.

**Comment by warrenbailey** [comment]
Benn has done the front end changes so this is ready for review now. Removing WIP label.

**Comment by warrenbailey** [comment]
Not ready now, we're going to add the transaction ID to the errors pages too

**Comment by warrenbailey** [comment]
Ready for review again

File: app/libs/utils.py [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
 def get_locale():
     return request.accept_languages.best_match(LANGUAGES.keys()) or 'cy'
 
 
+def convert_tx_id(tx_id):
+    # converts the guid tx_id to string of 16 characters with a space between every 4 characters
+    tx_id = uuid.UUID(tx_id)
+    tx_id = tx_id.hex
+    tx_id = tx_id.upper()
+    displayable_tx_id = (tx_id[i:i+4] for i in range(0, 16, 4))
```
Comment (dhilton) [inline app/libs/utils.py:15]: 👍 For the list comprehension 💃

**Comment by weapdiv-david** [comment]
All LGTM :+1:

**Pull request merged**